core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -98,95 +160,344 @@ public GeoBoundingBoxQueryBuilder bottomRight(String geohash) {
      * @param lon The longitude
      */
     public GeoBoundingBoxQueryBuilder bottomLeft(double lat, double lon) {
-        box[BOTTOM] = lat;
-        box[LEFT] = lon;
+        bottomRight.resetLat(lat);
+        topLeft.resetLon(lon);
         return this;
     }
 
+    /**
+     * Adds bottom left corner.
+     *
+     * @param point bottom left corner of bounding box.
+     */
     public GeoBoundingBoxQueryBuilder bottomLeft(GeoPoint point) {
         return bottomLeft(point.lat(), point.lon());
     }
 
+    /**
+     * Adds bottom left corner.
+     *
+     * @param point bottom left corner of bounding box.
+     */
     public GeoBoundingBoxQueryBuilder bottomLeft(String geohash) {
         return bottomLeft(GeoHashUtils.decode(geohash));
     }
 
     /**
-     * Adds top right point.
+     * Adds top right corner.
      *
-     * @param lat The latitude
-     * @param lon The longitude
+     * @param lat latitude of top right corner.
+     * @param lon longitude of top right corner.
      */
     public GeoBoundingBoxQueryBuilder topRight(double lat, double lon) {
-        box[TOP] = lat;
-        box[RIGHT] = lon;
+        topLeft.resetLat(lat);
+        bottomRight.resetLon(lon);
         return this;
     }
 
+    /**
+     * Adds top right corner.
+     *
+     * @param point topRight corner of bounding box.
+     */
     public GeoBoundingBoxQueryBuilder topRight(GeoPoint point) {
         return topRight(point.lat(), point.lon());
     }
 
+    /**
+     * Adds top right corner.
+     *
+     * @param geohash topRight corner of bounding box.
+     */
     public GeoBoundingBoxQueryBuilder topRight(String geohash) {
         return topRight(GeoHashUtils.decode(geohash));
     }
 
+    /**
+     * Specify whether or not to try and fix broken/wrapping bounding boxes.
+     * If set to true, also enables ignoreMalformed thus disabling geo point
+     * validation altogether.
+     **/
     public GeoBoundingBoxQueryBuilder coerce(boolean coerce) {
+        if (coerce) {
+            this.ignoreMalformed = true;
+        }
         this.coerce = coerce;
         return this;
     }
 
+    /** Returns whether or not to try and fix broken/wrapping bounding boxes. */
+    public boolean coerce() {
+        return this.coerce;
+    }
+
+    /**
+     * Specify whether or not to ignore validation errors of bounding boxes.
+     * Can only be set if coerce set to false, otherwise calling this
+     * method has no effect.
+     **/
     public GeoBoundingBoxQueryBuilder ignoreMalformed(boolean ignoreMalformed) {
-        this.ignoreMalformed = ignoreMalformed;
+        if (coerce == false) {
+            this.ignoreMalformed = ignoreMalformed;
+        }
```

Comment:
wondering about this logic, as well as the one anove in the coerce setter. I might have missed some previous discussion around it, but it seems like this change should happen rather than here? would that make sense?